Fix #3: Include all cookbooks, even if they don't have a recipe

adamfranco opened the pull request:
Cookbooks may provide resources that get executed and make changes to the node. 
This change includes all of the cookbooks, not only the ones with recipes in the run-list.

Fixes #3.

adamfranco committed:
Fix #3: Include all cookbooks, even if they don't have a recipe in the run-list.

Cookbooks may provide resources that get executed and make changes to the node. 
This change includes all of the cookbooks, not only the ones with recipes in the run-list.

irvingpop commented:
This is awesome and a super valuable point!  Thanks for finding that.

One thing I try to be careful about is changing default behavior on users.  Some users may prefer the current behavior, and we should preserve it as the default.

Could you please change this to be an argument to the resource?  something like:

``` ruby
cookbook_versions 'whatever' do
  all_cookbooks true
end
```

And then build a conditional into the provider for both types of behavior  (keeping false as the default).

Thank you!

adamfranco committed:
#3 Include all cookbooks is now an optional attribute and recipe.

adamfranco commented:
I've now added `all_cookbooks` as a recipe and restored the original operation as the default.  Additionally, I added an `all` recipe so that the `all_cookbooks` type of operation could easily be added to a node by adding `"recipe[cookbook_versions::all]"` to the `run_list`. Without this I think users would have to add a custom recipe in order to pass the attribute. I've also updated the README, so this should be good to go.

irvingpop commented:
Thank you for the awesome contribution!
